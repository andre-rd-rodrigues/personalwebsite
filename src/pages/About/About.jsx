import { motion, fadeInVariant } from "assets/motion/motionVariants";
import AnimatedHeading from "components/AnimatedHeading/AnimatedHeading";
import PageContainer from "components/PageContainer/PageContainer";
import { Col, Row } from "react-bootstrap";
import { useTranslation } from "react-i18next";
import styles from "./about.module.scss";
import profile from "assets/images/profile.png";
import ExperienceTimeline from "components/ExperienceTimeline/ExperienceTimeline";

const About = () => {
  const { t } = useTranslation();

  return (
    <PageContainer>
      <div className={styles.container}>
        <AnimatedHeading>
          <h1>{t("about_page.title")}</h1>
        </AnimatedHeading>
        <ExperienceTimeline />
        {/*   About me */}
        <motion.div
          variants={fadeInVariant}
          initial="hidden"
          whileInView="visible"
          className={`${styles.about} ${styles.section}`}
        >
          <Row>
            <Col lg={6} md={6} sm={12} className={styles.aboutTextContainer}>
              <h2 variants={fadeInVariant}>{t("about_page.about_me.title")}</h2>
              <p variants={fadeInVariant}>
                {t("about_page.about_me.description")} ðŸŽ¯
              </p>

              {/* Experience */}
              <div className={styles.section}>
                <motion.h2
                  variants={fadeInVariant}
                  style={{ marginTop: "30px" }}
                >
                  {t("about_page.experience.title")}
                </motion.h2>
                <motion.ul variants={fadeInVariant}>
                  <li>{t("about_page.experience.web_basics")}</li>
                  <li>{t("about_page.experience.web_advanced")}</li>
                  <li>{t("about_page.experience.business")}</li>
                  <li>{t("about_page.experience.arquitecture")}</li>
                  <li>{t("about_page.experience.seo")}</li>
                  <li>{t("about_page.experience.analytics")}</li>
                  <li>{t("about_page.experience.responsive")}</li>
                  <li>{t("about_page.experience.testing")}</li>
                </motion.ul>
              </div>

              {/* Education */}
              <div className={styles.educationContainer}>
                <motion.h2
                  variants={fadeInVariant}
                  style={{ marginTop: "30px" }}
                >
                  {t("about_page.education")}
                </motion.h2>
                <motion.ul variants={fadeInVariant}>
                  <li>
                    <a href="img/docs/c_frontend.png" download="">
                      JavaScript Algorithms and Data Structures
                    </a>
                  </li>
                  <li>
                    <a href="img/docs/c_frontend.png" download="">
                      Front-End Libraries Certification
                    </a>
                  </li>
                  <li>
                    <a href="img/docs/c_backend.png" download="">
                      Back-End Development and APIs Certification
                    </a>
                  </li>
                  <li>The Complete Node.js Course | Code with Mosh</li>
                  <li>The Ultimate Redux Course | Code with Mosh</li>
                  <li>The Ultimate Docker Course | Code with Mosh</li>
                  <li>
                    <a href="img/docs/c_algorithms.png" download="">
                      Algorithms and Data Structures Certification
                    </a>
                  </li>
                  <li>
                    <a href="img/docs/c_responsive.png" download="">
                      Responsive Web Design Certification
                    </a>
                  </li>
                  <li>Design Thinking Certification</li>
                  <li>UX/UI Fundamentals Certification</li>
                </motion.ul>
              </div>
            </Col>
            <Col lg={6} md={6} sm={12} className={styles.profileImageContainer}>
              <img
                alt="profile_picture"
                src={profile}
                className={styles.profileImage}
                loading="lazy"
              />
            </Col>
          </Row>
        </motion.div>
      </div>
    </PageContainer>
  );
};

export default About;
