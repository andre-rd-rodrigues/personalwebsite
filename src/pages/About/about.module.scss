@import "assets/styles/mixins";

.container {
  h1 {
    font-size: var(--fs-xxl);
    max-width: 80vw;
  }
}

@include title_transparent;

.section {
  margin: 60px 0;
  p {
    max-width: 600px;
  }
  h2 {
    margin-bottom: 30px;
    font-family: Jost, sans-serif;
    text-transform: uppercase;
  }
  p,
  h2 {
    font-size: var(--fs-m);
  }
  span {
    display: block;
    width: 100%;
    height: 20px;
  }
}

//Services
.services {
  h2 {
    &:last-of-type {
      margin-top: 50px;
    }
  }
  h3 {
    font-family: Italiana;
    font-size: 1.8rem;
  }
}
.serviceCol {
  display: flex;
  justify-content: center;
}
//About
.aboutTextContainer {
  a {
    text-decoration: dotted;
    text-decoration-color: rgba(255, 255, 255, 0.747);
    text-decoration-line: underline;
    transition: 0.3s ease;
    &:hover {
      color: $yellow;
    }
  }
}

//Education
.educationContainer {
  margin-top: 50px;
  ul {
    list-style: none;

    li {
      margin: 20px 0;
      position: relative;

      //Dot
      &::after {
        position: absolute;
        top: 10px;
        left: -20px;
        content: "";
        background-color: white;
        width: 6px;
        height: 6px;
        border-radius: 50%;
      }
    }
  }
  a {
    color: inherit;
  }
}

// Image
.profileImage {
  width: 100%;
  max-width: 500px;
  height: 100%;
  background: url("../../assets/images/profile.png") center center no-repeat;
  background-size: cover;
  margin: auto;
}

//Responsive
@media only screen and (max-width: $breaking-point-md) {
  .about {
    text-align: start;
  }
  .aboutTextContainer {
    align-items: flex-start;
    order: 2;
  }
  .educationContainer {
    //Dot
    ul li::after {
      position: absolute;
      top: 10px;
      left: -30px;
      content: "";
      background-color: white;
      width: 6px;
      height: 6px;
      border-radius: 50%;
    }
  }
  .profileImage {
    min-width: 100%;
    min-height: 300px;
    background-position: 0 -20vw;
  }

  .profileImageContainer {
    order: 1;

    //Same value as .section
    margin-bottom: 50px;
  }
  .serviceCol {
    display: flex;
    justify-content: center;
  }
}
