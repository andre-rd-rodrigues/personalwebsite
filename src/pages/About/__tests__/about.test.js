import { screen } from "@testing-library/react";
import { render } from "utils/tests/test-utils";

import About from "../About";

describe("About page", () => {
  it("renders About me title", () => {
    render(<About />);

    const titleElement = screen.getByText(/About me/i);

    expect(titleElement).toBeInTheDocument();
  });

  it("renders page correctly", () => {
    const { asFragment } = render(<About />);

    expect(asFragment()).toMatchSnapshot();
  });
});
