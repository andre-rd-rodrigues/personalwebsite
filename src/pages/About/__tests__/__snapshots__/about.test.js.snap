// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About page renders page correctly 1`] = `
<DocumentFragment>
  <div
    class="container undefined"
  >
    <div
      class="wrapper"
      style="opacity: 0;"
    >
      <div
        class="container"
      >
        <div
          class="container undefined"
        >
          <span
            style="white-space: nowrap;"
          >
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                W
              </h1>
            </span>
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                e
              </h1>
            </span>
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                b
              </h1>
            </span>
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                 
              </h1>
            </span>
          </span>
          <span
            style="white-space: nowrap;"
          >
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                d
              </h1>
            </span>
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                e
              </h1>
            </span>
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                v
              </h1>
            </span>
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                e
              </h1>
            </span>
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                l
              </h1>
            </span>
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                o
              </h1>
            </span>
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                p
              </h1>
            </span>
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                e
              </h1>
            </span>
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                r
              </h1>
            </span>
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                 
              </h1>
            </span>
          </span>
          <span
            style="white-space: nowrap;"
          >
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                p
              </h1>
            </span>
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                a
              </h1>
            </span>
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                s
              </h1>
            </span>
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                s
              </h1>
            </span>
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                i
              </h1>
            </span>
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                o
              </h1>
            </span>
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                n
              </h1>
            </span>
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                a
              </h1>
            </span>
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                t
              </h1>
            </span>
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                e
              </h1>
            </span>
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                 
              </h1>
            </span>
          </span>
          <span
            style="white-space: nowrap;"
          >
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                a
              </h1>
            </span>
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                b
              </h1>
            </span>
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                o
              </h1>
            </span>
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                u
              </h1>
            </span>
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                t
              </h1>
            </span>
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                 
              </h1>
            </span>
          </span>
          <span
            style="white-space: nowrap;"
          >
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                i
              </h1>
            </span>
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                n
              </h1>
            </span>
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                n
              </h1>
            </span>
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                o
              </h1>
            </span>
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                v
              </h1>
            </span>
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                a
              </h1>
            </span>
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                t
              </h1>
            </span>
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                i
              </h1>
            </span>
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                o
              </h1>
            </span>
            <span
              style="overflow: hidden; display: inline-block;"
            >
              <h1
                style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
              >
                n
              </h1>
            </span>
          </span>
        </div>
        <div
          class="about section"
          style="opacity: 0;"
        >
          <div
            class="row"
          >
            <div
              class="aboutTextContainer col-lg-6 col-md-6 col-sm-12"
            >
              <h2
                variants="[object Object]"
              >
                About me
              </h2>
              <p
                variants="[object Object]"
              >
                Ever since I decided to make my passion for technology a professional career, I have been fully dedicated to the development of digital projects and the acquisition of new knowledge, in order to add competitive value in the creation of digital solutions.
My attention to detail, creative vision of the world and meticulous nature are the main contributions of my identity to the development and implementation of custom solutions. Throughout my career, I have developed projects for various sectors and I am mainly dedicated to creating websites, web apps, and branding.
My primary goal is to establish quick and long-lasting solutions, relationships, and assist in developing and implementing high-quality products. 🎯
              </p>
              <div
                class="section"
              >
                <h2
                  style="margin-top: 30px; opacity: 0;"
                >
                  Experience
                </h2>
                <ul
                  style="opacity: 0;"
                >
                  <li>
                    Created and optimized projects using web basics (HTML, CSS, and SASS) and modern JavaScript libraries and frameworks like React.js, React Native, Next JS, Typescript and adjusting advanced state-management with Redux and React Context;
                  </li>
                  <li>
                    Improved skills in back-end, creating API's with Node JS, connecting databases with MongoDB and using Docker to deploy and consume locally through RESTfull API requests.
                  </li>
                  <li>
                    Structured, applied, and enhanced front-end architecture and data-structure to improve efficiency and sustainability, tailored to the business area;
                  </li>
                  <li>
                    Skilled in using microservices and event-driven architecture to develop and maintain features, and taking ownership of the development process from design and code review to implementation;
                  </li>
                  <li>
                    Skilled in using SEO optimization techniques, such as pre-rendering, server-side rendering, keyword usage and proper page indexing for maximizing the volume of inbound organic traffic from search engines to the website;
                  </li>
                  <li>
                    Skilled in using advanced analytics tools like Google Analytics and Insights, configuring and customizing the tools appropriately, identifying gaps in existing implementations and recommending solutions to enhance data collection, promote scalability, and drive best practices;
                  </li>
                  <li>
                    Standardized all output with a new, responsive, mobile-first-approach through advanced responsive web design and UX/UI best practices through Bootstrap, Tailwind CSS, media queries, responsive layouts and performance best pratices;
                  </li>
                  <li>
                    Experience with testing frameworks, performance testing, and unit testing with Jest, React Testing Library and Cypress.
                  </li>
                </ul>
              </div>
              <div
                class="educationContainer"
              >
                <h2
                  style="margin-top: 30px; opacity: 0;"
                >
                  Education
                </h2>
                <ul
                  style="opacity: 0;"
                >
                  <li>
                    <a
                      download=""
                      href="img/docs/c_frontend.png"
                    >
                      JavaScript Algorithms and Data Structures
                    </a>
                  </li>
                  <li>
                    <a
                      download=""
                      href="img/docs/c_frontend.png"
                    >
                      Front-End Libraries Certification
                    </a>
                  </li>
                  <li>
                    <a
                      download=""
                      href="img/docs/c_backend.png"
                    >
                      Back-End Development and APIs Certification
                    </a>
                  </li>
                  <li>
                    The Complete Node.js Course | Code with Mosh
                  </li>
                  <li>
                    The Ultimate Redux Course | Code with Mosh
                  </li>
                  <li>
                    The Ultimate Docker Course | Code with Mosh
                  </li>
                  <li>
                    <a
                      download=""
                      href="img/docs/c_algorithms.png"
                    >
                      Algorithms and Data Structures Certification
                    </a>
                  </li>
                  <li>
                    <a
                      download=""
                      href="img/docs/c_responsive.png"
                    >
                      Responsive Web Design Certification
                    </a>
                  </li>
                  <li>
                    Design Thinking Certification
                  </li>
                  <li>
                    UX/UI Fundamentals Certification
                  </li>
                </ul>
              </div>
            </div>
            <div
              class="profileImageContainer col-lg-6 col-md-6 col-sm-12"
            >
              <div
                class="profileImage"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
